common/gera/map.go

```diff
@@ -29,79 +29,130 @@
 package gera
 
 import (
+	"sync"
+
 	"dario.cat/mergo"
 )
 
-type Map interface {
-	Wrap(m Map) Map
+type Map[K comparable, V any] interface {
```

Comment:
is there a reason why use type arguments? I see only Map[string,string] used everywhere.

Reply:
It would probably be safe to make the key a string forever, not so sure about the value. Basically `Map` already existed and I didn't want to preclude using non-string values in the future, if we so wish, mostly for the purposes of the template system, as well as using `gera` in contexts other than the template system. Since Go got generics a few years ago, the untemplated `gera.Map` became old style go, relying on `interface{}`, so the paths forward were either (1) to remove it and only use `gera.StringMap`, or (2) to update `gera.Map` for modern Go. I choose 2.

Reply:
okay, thank you